src/shred/shred.rs

```diff
@@ -0,0 +1,450 @@
+#![crate_name = "uu_shred"]
+
+/*
+* This file is part of the uutils coreutils package.
+*
+* (c) Fort <forticulous@gmail.com>
+*
+* For the full copyright and license information, please view the LICENSE
+* file that was distributed with this source code.
+*/
+
+extern crate getopts;
+extern crate rand;
+
+use rand::{ThreadRng, Rng};
+use std::cell::{Cell, RefCell};
+use std::fs;
+use std::fs::File;
+use std::io;
+use std::io::prelude::*;
+use std::path::{Path, PathBuf};
+
+#[macro_use]
+extern crate uucore;
+
+static NAME: &'static str = "shred";
+static VERSION_STR: &'static str = "1.0.0";
+const BLOCK_SIZE: usize = 512;
```

Comment:
> > +const BLOCK_SIZE: usize = 512;
> 
> I looked into MetadataExt::blksize(), it looks like it's only for unix file systems

Does shred do anything useful on NTFS? If not we can just make it a unix
only util.